collector/zfs.go

```diff
@@ -35,9 +35,25 @@ type zfsSubsystemName string
 
 const (
 	arc            = zfsSubsystemName("zfsArc")
```

Comment:
Looks like we missed this in the initial ZFS PR, but all these shouldn't be global either. I also think all those types aren't necessary, so instead wherever those consts are used we can just use the strings. Unless I miss a reason why we shouldn't.
If you feel like cleaning this up, that would be great. But we can also ignore this here for now and I'll clean it up in another PR.